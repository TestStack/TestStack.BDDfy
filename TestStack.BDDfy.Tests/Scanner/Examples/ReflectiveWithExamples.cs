using ApprovalTests;
using NUnit.Framework;
using Shouldly;
using TestStack.BDDfy.Reporters;

namespace TestStack.BDDfy.Tests.Scanner.Examples
{
    [TestFixture]
    public class ReflectiveWithExamples
    {
        public string SecondExample { get; set; }

        public void GivenStepWith__FirstExample__PassedAsParameter(int firstExample)
        {
            firstExample.ShouldBeOneOf(1, 2);
        }

        public void GivenStepWith__SecondExample__AccessedViaProperty()
        {
            SecondExample.ShouldBeOneOf("foo", "bar");
        }

        [Test]
        public void Run()
        {
            var story = this
                .WithExamples(new ExampleTable("First Example", "Second Example")
                {
                    {1, "foo"},
                    {2, "bar"}
                })
                .BDDfy();

            var reporter = new TextReporter();
            reporter.Process(story);
            Approvals.Verify(reporter.ToString());
        }
    }
}