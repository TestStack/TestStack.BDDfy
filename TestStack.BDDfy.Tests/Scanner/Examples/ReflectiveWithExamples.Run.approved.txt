
Scenario: Reflective with examples
	Given step with <first example> passed as parameter   
	Given step with <second example> accessed via property

Examples: 
| First Example | Second Example |
| 1             | foo            |
| 2             | bar            |

